src/srcShared/fs.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+
+const fs = vscode.workspace.fs
+type Uri = vscode.Uri
+
+/**
+ * @warning Do not import this class specifically, instead import the instance {@link fsCommon}.
+ *
+ * This class contains file system methods that are "common", meaning
+ * it can be used in the browser or desktop.
+ *
+ * Technical Details:
+ * TODO: Verify the point below once I get this hooked up to the browser code.
+ * - vscode.workspace.fs dynamically resolves the correct file system provider
+ *   to use. This means that in the browser it will attempt to use the File System
+ *   Access API.
+ *
+ *
+ * MODIFYING THIS CLASS:
+ * - All methods must work for both browser and desktop
+ * - Do not use 'fs' or 'fs-extra' since they are not browser compatible.
+ *   If they have functionality that cannot be achieved with a
+ *   browser+desktop implementation, then create it in the module {@link TODO}
+ */
+export class FileSystemCommon {
+    constructor() {}
+
+    async mkdir(path: Uri | string): Promise<void> {
+        path = FileSystemCommon.getUri(path)
```

Comment:
In general we probably want to force callers to provide a Uri. However this fs module is probably used in many legacy places so we probably need to support string names as you have done.

Reply:
Yeah, this is to support legacy for now. It will be one of those things where in PR review we'll want to ensure a `vscode.Uri` object instead of a string.